### ISSUE TITLE
Nested fields in compound @keys raise Unexpected end of JSON input

### ISSUE THREAD

**Issue by maciejwisniowski** [wundergraph/graphql-go-tools#743]
We're using the Cosmo Router 0.63.1 (the latest so far) with Federation and two subgraphs defined as:

Subgraph one:

```
type Account @key(fields: "id") {
  id: ID!
}

type User @key(fields: "id account { id }") {
  id: ID!
  account: Account!
  name: String!
}

type Query {
  test: User!
}
```

Subgraph two:
```
type Account @key(fields: "id") {
  id: ID! @external
}

type User @key(fields: "id account { id }") @extends {
  id: ID! @external
  account: Account! @external
  foo: Boolean!
}
```

Running the following query raises exception:

`query { test{ id name foo } }`

**Unexpected end of JSON input** in the GraphQL UI and **index out of range [5] with length 3** in the server's log.

This seems to be related to AST building phase so I assume the problem is with graphql-go-tools.

The same subgraph works well with the ApolloGateway/Federation. Without the nested compound key it also works with Cosmo Router.
 We've created a minimal docker-compose setup to reproduce the issue. It is available at: https://github.com/maciejwisniowski/wg-router-bug

The Router's backtrace is attached
[wg-router-bt.json](https://github.com/wundergraph/graphql-go-tools/files/14221653/wg-router-bt.json)

**Comment by jensneuse** [comment]
Thank you for this excellent bug report including a reproducible. Forwarding to the team to investigate once we have time. I'll update this thread.

**Comment by Aenimus** [comment]
Hi @maciejwisniowski,

This is actually an issue with the configuration passed by the router. External fields that appear in FieldSets are added to the fieldNames property of root nodes. There will be a patch to fix this. I'll keep you updated.

**Comment by devsergiy** [comment]
Hi @maciejwisniowski 

fix is included beginning from version [0.67.0](https://github.com/wundergraph/cosmo/compare/router@0.66.3...router@0.67.0)
